File: sanghoon/programmers-2/전화번호 목록.py
def solution(phone_book):
    phone_book.sort()

    for i in range(len(phone_book) - 1):
        if phone_book[i + 1].startswith(phone_book[i]):
            return False

    return True

phone_book = ["12", "123", "1235", "567", "88"]
solution(phone_book)

File: kyulee/programmers-2/전화번호-목록.java
package org.example;

import java.util.*;

public class 전화번호-목록 {
    public static boolean solution(String[] phone_book) {

        Map<String, Integer> numberList = new HashMap<>();

        for (int i = 0; i < phone_book.length; i++) {
            numberList.put(phone_book[i], i);
        }


        for (int i = 0; i < phone_book.length; i++){
            for (int j = 0; j < phone_book[i].length(); j++) {
                if (numberList.containsKey(phone_book[i].substring(0, j))) {
                    return false;
                }
            }
        }
        return true;
    }

    public static void main(String[] args) {
        String[] phoneBook = {"119", "97674223", "1195524421"};

        boolean result = solution(phoneBook);
        System.out.println(result);
    }
}
